client/src/app/templates/view/template-view.component.ts

```diff
@@ -0,0 +1,54 @@
+import { Component, OnInit } from '@angular/core';
+import { TemplateService, ToasterService } from '../../_services';
+import { ActivatedRoute, Router } from '@angular/router';
+import { Template, Service } from '../../_models';
+import { MatDialog } from '@angular/material';
+import { ConfirmationDialogComponent } from '../../dialogs/confirmation-dialog/confirmation-dialog.component';
+
+@Component({
+    templateUrl: './template-view.component.html',
+})
+
+export class TemplateViewComponent implements OnInit {
+
+    template: Template = new Template;
+    services: Service[] = new Array<Service>();
+
+
+
+    constructor(
+        private templateService: TemplateService,
+        private toasterService: ToasterService,
+        private route: ActivatedRoute,
+        private router: Router,
+        public dialog: MatDialog
+    ) {}
+
+    ngOnInit() {
+        this.route.params.subscribe(params => {
+            this.template.id = params['id'];
+            this.templateService.getById(+this.template.id).subscribe(template => {
+                this.template = template;
+            });
+        });
+        this.changeFormatDate();
+    }
+
+    toDate(val) {
+        const date = new Date();
+        date.setDate(val.day);
+        date.setHours(val.hours);
+        date.setMinutes(val.minutes);
+        return date;
+    }
+
+    changeFormatDate() {
+        this.services = this.template.services.map(n =>
```

Comment:
here @Embraser01 `cannot read property 'map' of undefined ` et après la même  `cannot read property 'day' of undefined `

Reply:
Je pense que c'est dû au fait que tu fasses un `new Template` ligne 14, ça initialise un objet mais le tableau `services` est pas initialisé

Reply:
Tu dois vérifier que le template est chargé avant de faire ça ^^

Reply:
Est ce que je peux régler ça en mettant un await avant la récupération du template ligne 28 ?

Reply:
Nop je pense pas que ça marche, fait peut-être plus un check au début
de changeFormatDate()

Reply:
Mais si je fais un check et qu'il me répond que le template est toujours pas chargé ben il va jamais effectuer ma conversion en Date non?

Reply:
Enfait tu as juste a appeler `changeFormatDate()` dans le subscribe à la place de la ligne 34